pkg/xen-tools/initrd/udhcpc_script.sh

```diff
@@ -11,20 +11,28 @@
 : "${hostname:=}"
 : "${router:=}"
 : "${mask:=}"
+: "${dns:=}"
 
 [ -z "$1" ] && echo 'Error: should be called from udhcpc' && exit 1
 
+# Nameservers collected separately per interface, then combined together
+# into resolv.conf of the container app.
+VM_RESOLV_CONF_DIR="/run/resolvconf"
+mkdir -p "${VM_RESOLV_CONF_DIR}"
+
 # create etc directory if not already done
 mkdir -p /mnt/rootfs/etc
 
 # save config information for $interface
 CFG="/mnt/rootfs/etc/udhcpc.${interface}.cfg"
 
+TMP_RESOLV_CONF='/mnt/rootfs/tmp/resolv.conf'
```

Comment:
Is that in the same filesystem as RESOLVE_CONF or in a tmpfs?
A rename across file systems turns into a cp+rm hence is not dynamic.

Reply:
Good catch, it is indeed a different filesystem.
I have moved the temporary file right next to the destination resolv.conf (same dir).